storage/provider/managedfs.go

```diff
@@ -240,7 +241,7 @@ func mountFilesystem(run runCommandFunc, dirFuncs dirFuncs, devicePath, mountPoi
 	if mtabEntry == "" {
 		return nil
 	}
-	return addFstabEntry(etcDir, devicePath, mountPoint, mtabEntry)
+	return addFstabEntry(etcDir, devicePath, UUID, mountPoint, mtabEntry)
```

Comment:
we probably should only do this `once` to update the dev path entry for the mounted scenario.
Also `ensureFstabEntry` might be a better name now.

Reply:
renamed found to addNewEntry

Reply:
also, we do only update fstab on first attach and then again a second time only if the UUID changes from "" to "something".